src/plugins/intel_gpu/src/kernel_selector/kernels/dynamic_quantize/dynamic_quantize_kernel_opt.cpp

```diff
@@ -0,0 +1,159 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "dynamic_quantize_kernel_opt.h"
+#include "kernel_selector_utils.h"
+#include <string>
+
+
+static constexpr size_t simd = 16;
+
+namespace kernel_selector {
+static std::pair<size_t, size_t> get_input_bf_size(const dynamic_quantize_params& params) {
+    size_t input_f = params.inputs[0].Feature().v;
+    size_t input_batch = params.inputs[0].Batch().v;
+    // 3D input
+    if (params.outputs[0].GetLayout() == DataLayout::bfyx) {
+        input_f = params.inputs[0].Y().v * params.inputs[0].X().v;
+        input_batch = params.inputs[0].Batch().v * params.inputs[0].Feature().v;
+    }
+
+    return {input_batch, input_f};
+}
+
+static size_t get_match_vector_size(const dynamic_quantize_params& params) {
+    auto block_sizes = { 8, 4, 2 };
+
+    for (auto block_size : block_sizes) {
+        if (((params.inputs[0].X().v * params.inputs[0].Y().v) / simd) % block_size == 0) {
+            return block_size;
+        }
+    }
+
+    return 1;
+}
+
+ParamsKey DynamicQuantizeKernelOpt::GetSupportedKey() const {
+    ParamsKey k;
+    k.EnableInputDataType(Datatype::F16);
+    k.EnableOutputDataType(Datatype::INT8);
+    k.EnableDifferentTypes();
+    k.EnableAllInputLayout();
+    k.EnableAllOutputLayout();
+    k.EnableTensorOffset();
+    k.EnableTensorPitches();
+    k.EnableBatching();
+    k.EnableDynamicShapesSupport();
+    return k;
+}
+
+JitConstants DynamicQuantizeKernelOpt::GetJitConstants(const dynamic_quantize_params& params) const {
+    JitConstants jit = MakeBaseParamsJitConstants(params);
+
+    auto vec_size = get_match_vector_size(params);
+    auto bf_size = get_input_bf_size(params);
+    auto total_block_num = bf_size.second / (simd * vec_size);
+    size_t aligned_block_num = (total_block_num > 32) ? Align(total_block_num, 32) : total_block_num;
+    size_t block_num = (total_block_num > 32) ? 32 : total_block_num;
+
+    jit.AddConstant(MakeJitConstant("VEC_SIZE", vec_size));
+    jit.AddConstant(MakeJitConstant("SIMD", simd));
+    jit.AddConstant(MakeJitConstant("TOTAL_BLOCK_NUM", total_block_num));
+    jit.AddConstant(MakeJitConstant("ALIGNED_BLOCK_NUM", aligned_block_num));
+    jit.AddConstant(MakeJitConstant("BLOCK_NUM", block_num));
+    jit.Merge(GetTensorFriendlyWorkGroupsJit(params.outputs[0]));
+
+    GPU_DEBUG_TRACE_DETAIL << "DynamicQuantizeKernelOpt VEC_SIZE(" << vec_size << ") input bfyx (" << params.inputs[0].Batch().v
+            << ", " << params.inputs[0].Feature().v << ", " << params.inputs[0].Y().v << ", "  << params.inputs[0].X().v << ")" << std::endl;
```

Comment:
I'd say that this is redundant print as one can always dump sources and see jitted values

Reply:
applied, thanks!